### PR TITLE
Introduce Java 8 for URI

### PR DESCRIPTION
**Pull request by knutwalker**
This is more of an educational PR.

This is be my approach to write URI using more
functional and Java 8 features (acutally, only Optional).

### CODE REVIEW

**Comment by boferri** [comment]
nice, but afaik, we cannot rely on Java 8 here, because we also make use of dswarm-commons in the dswarm unmanaged extension for Neo4j. So I think that we need to go with Java 7 in dswarm-commons until Neo4j goes (officially) Java 8 and we can upgrade the Java level at the dswarm unmanaged extension for Neo4j as well.

File: src/main/java/org/dswarm/common/web/URI.java [hunk 1]
```diff
@@ -15,75 +15,74 @@
...
 
-		return uriPart != null && !uriPart.trim().isEmpty();
+	public static URI create(final String uriString) {
+		return Optional
+				.ofNullable(uriString)
+				.map(URI::parse)
+				.orElseGet(() -> new URI(Optional.empty(), Optional.empty()));
 	}
 
-	@Override public String toString() {
```
Comment (boferri) [inline src/main/java/org/dswarm/common/web/URI.java:85]: toString is missing now, or? - should return the full uri.

Comment (knutwalker) [inline src/main/java/org/dswarm/common/web/URI.java:85]: It's a few lines above :-)

**Comment by knutwalker** [comment]
Ah, OK, I forgot about the extension. Well, I guess that concludes it then. But as I said, this was more educational anyway :-)

File: src/main/java/org/dswarm/common/web/URI.java (hunk 1 above)
Comment (boferri) [inline src/main/java/org/dswarm/common/web/URI.java:85]: damn -.-

**Comment by boferri** [comment]
yepp, we can keep this for later. thanks a lot.

**Pull request closed without merging**